### PR TITLE
Added Windows Aggregation Definition in CDAP

### PR DESCRIPTION
**Pull request by udurbha**
As part of Bigquery Pushdown, where instead spark sql function calls ,bigquery function calls are executed,added windows aggregation definition.Followed by Bigquery Windows aggregation function calls implementation in google-cloud repo and transform method will be in WIndows-aggregation repo.

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/WindowsAggregationDefinition.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+     */
+    public Expression getExpression() {
+        return expression;
+    }
+
+        /**
+         *
+         * @return the function applied to select a single record from duplicate records
+         */
+        public FilterFunction getFilterFunction() {
```
Comment (fernst) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/WindowsAggregationDefinition.java:86]: Spacing is off here.

File: cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/WindowsAggregationDefinition.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    /**
+     * Builds a WindowsAggregationDefinition using fields to group by and fields to select.
+     * The fields to select must be specified, while the fields to group by are optional.
+     */
+    public static class Builder {
+        private List<Expression> windowsExpressions;
+        private Map<String, Expression> selectExpressions;
+
+        public Builder() {
+            windowsExpressions = Collections.emptyList();
```
Comment (fernst) [inline cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/aggregation/WindowsAggregationDefinition.java:117]: I don't think these are mutable instances. You need to initialize them using new ArrayList() and new HashMap();

File: cdap-features/src/main/java/io/cdap/cdap/features/Feature.java [hunk 3]
```diff
@@ -31,6 +31,7 @@ public enum Feature {
   PIPELINE_COMPOSITE_TRIGGERS("6.8.0"),
   PUSHDOWN_TRANSFORMATION_GROUPBY("6.7.0"),
   PUSHDOWN_TRANSFORMATION_DEDUPLICATE("6.7.0"),
+  PUSHDOWN_TRANSFORMATION_WINDOWSAGGREGATION("6.8.0"),
```
Comment (fernst) [inline cdap-features/src/main/java/io/cdap/cdap/features/Feature.java:34]: this would be for 6.9.0

**Review by fernst — CHANGES_REQUESTED** [review]
Added a few comments/suggestions.

**Comment by udurbha** [comment]
PR was sent before Design Doc.

**Pull request closed without merging**